embrace-android-core/src/test/java/io/embrace/android/embracesdk/internal/envelope/log/LogEnvelopeSourceImplTest.kt

```diff
@@ -1,57 +1,82 @@
 package io.embrace.android.embracesdk.internal.envelope.log
 
+import io.embrace.android.embracesdk.fakes.FakeCachedLogEnvelopeStore
 import io.embrace.android.embracesdk.fakes.FakeEnvelopeMetadataSource
 import io.embrace.android.embracesdk.fakes.FakeEnvelopeResourceSource
 import io.embrace.android.embracesdk.fakes.FakeLogPayloadSource
+import io.embrace.android.embracesdk.fakes.fakeEnvelopeMetadata
+import io.embrace.android.embracesdk.fakes.fakeEnvelopeResource
+import io.embrace.android.embracesdk.fixtures.nativeCrashLog
+import io.embrace.android.embracesdk.fixtures.nativeCrashWithoutSessionLog
 import io.embrace.android.embracesdk.fixtures.sendImmediatelyLog
+import io.embrace.android.embracesdk.fixtures.testLog
+import io.embrace.android.embracesdk.internal.delivery.PayloadType
+import io.embrace.android.embracesdk.internal.delivery.SupportedEnvelopeType
+import io.embrace.android.embracesdk.internal.delivery.storage.CachedLogEnvelopeStore.Companion.createNativeCrashEnvelopeMetadata
 import io.embrace.android.embracesdk.internal.logs.LogRequest
+import io.embrace.android.embracesdk.internal.opentelemetry.embProcessIdentifier
 import io.embrace.android.embracesdk.internal.payload.LogPayload
+import io.embrace.android.embracesdk.internal.spans.findAttributeValue
+import io.opentelemetry.semconv.incubating.SessionIncubatingAttributes
 import org.junit.Assert.assertEquals
 import org.junit.Assert.assertTrue
 import org.junit.Before
 import org.junit.Test
 
 internal class LogEnvelopeSourceImplTest {
```

Comment:
These tests were quite wrong before so they were rewritten to verify the right things, plus what was added in this PR